tests/block_header_test.go

```diff
@@ -155,3 +184,89 @@ func testHeaders_BaseFeeEvolutionFollowsPricingRules(t *testing.T, headers []*ty
 		)
 	}
 }
+
+func testHeaders_TransactionRootMatchesBlockTxsHash(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		block, err := client.BlockByNumber(context.Background(), header.Number)
+		require.NoError(err, "failed to get block receipts")
+
+		txsHash := types.DeriveSha(block.Transactions(), trie.NewStackTrie(nil))
+		require.Equal(header.TxHash, txsHash, "transaction root hash mismatch")
+	}
+}
+
+func testHeaders_ReceiptRootMatchesBlockReceipts(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(header.Number.Uint64())))
+		require.NoError(err, "failed to get block receipts")
+
+		receiptsHash := types.DeriveSha(types.Receipts(receipts), trie.NewStackTrie(nil))
+		require.Equal(header.ReceiptHash, receiptsHash, "receipt root hash mismatch")
+	}
+}
+
+func testHeaders_LogsBloomMatchesLogsInReceipts(t *testing.T, headers []*types.Header, client *ethclient.Client) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		receipts, err := client.BlockReceipts(context.Background(),
+			rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(header.Number.Uint64())))
+		require.NoError(err, "failed to get block receipts")
+
+		logsBloom := types.CreateBloom(receipts)
+		require.Equal(header.Bloom, logsBloom, "logs bloom mismatch")
+	}
+}
+
+func testHeaders_CoinbaseIsZeroForAllBlocks(t *testing.T, headers []*types.Header) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		require.Empty(header.Coinbase, "coinbase is not zero")
+	}
+}
+
+func testHeaders_DifficultyIsZeroForAllBlocks(t *testing.T, headers []*types.Header) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		// Cmp returns 0 when the values are equal
+		require.Equal(0, big.NewInt(0).Cmp(header.Difficulty), "difficulty is not zero")
+	}
+}
+
+func testHeaders_NonceIsZeroForAllBlocks(t *testing.T, headers []*types.Header) {
+	require := require.New(t)
+
+	for _, header := range headers {
+		require.Empty(header.Nonce.Uint64(), "nonce is not zero")
+	}
+}
+
+func testHeaders_TimeProgressesMonotonically(t *testing.T, headers []*types.Header) {
+	require := require.New(t)
+
+	for i := 1; i < len(headers); i++ {
+		require.GreaterOrEqual(headers[i].Time, headers[i-1].Time, "time is not monotonically increasing")
```

Comment:
Nit: we do also store the nano-seconds in the block header, enabling us to get the block time at nano-second granularity (see [here](https://github.com/Fantom-foundation/Sonic/blob/2001b3718c9182b159897adc0ceb66ba608dcb47/tests/block_header_test.go#L113-L118)). Can you check that this time in each block is strictly greater than in its predecessor?